README.md

```diff
@@ -1,14 +1,42 @@
-# gin-in-a-can
+# can
 
-[The existing openapi generator for Gin](https://openapi-generator.tech/docs/generators/go-gin-server)  is decent, but it lacks certain features that I need.
-Ideally, I would contribute to it directly, but I need this sooner than I am able to learn how their build system, etc. work.
+The gin templates that are available in the [existing openapi generator](https://openapi-generator.tech/docs/generators/go-gin-server)
+require in place modification. This does not suit our workflow. We would like to be able to regenerate our API interface
+without having to patch in our own code every time. gRPC generates code that illustrates how to generate code that can
+be used without modification.
 
-The features I need are:
-* To be able to register an existing gin Engine with predefined middleware.
-* To be able to use the generated code without modifying it.
+This repository is intended to apply the pattern used by gRPC to Gin for rest services.
 
-## TODO
-* use components to avoid duplicate ref structs
-* use nullable types in parameters and request bodies
-* Add parameter validation based on the `required` flag, regex patterns and other OpenAPI format specifications
-* Add API Fuzz testing
\ No newline at end of file
+## Usage
+Download the latest [release](https://github.com/SasSwart/gin-in-a-can/releases) archive for your platform and extract it.
+Copy the [example config file](cmd/petstore/can.yml) to a convenient location and modify it to your needs.
+
+Run the can command as follows:
+`/path/to/can --configFile=/path/to/can.yml`
+
+## Design
+See [design.md](design.md)
+
+## Roadmap
+
+### TODOs
+* Look into yaml.Readpath
+* Look into unmarshalling composed types
+* Implement e2e testing of generated code (1.0)
+* (Alex) test resolveRefs against nested refs and debug - use hackday branch for reference
```

Comment:
Could be misleading, putting this note here to create awareness and record of a pending update, not blocking though